presto-main/src/main/java/com/facebook/presto/metadata/BuiltInFunctionNamespaceManager.java

```diff
@@ -901,6 +904,14 @@ private SpecializedFunctionKey doGetSpecializedFunctionKey(Signature signature)
         Iterable<SqlFunction> candidates = getFunctions(null, signature.getName());
         // search for exact match
         Type returnType = typeManager.getType(signature.getReturnType());
+
+        if (returnType instanceof EnumType && CastType.CAST.getCastName().equals(signature.getName())) {
```

Comment:
Why is this logic in BuiltInFunctionNamespaceManager?

Reply:
I reorganized it a bit in #14728 to have the enum-specific logic only live in EnumOperators (but we still have a call from BuiltinFunctionNamespaceManager - I think it makes sense to have logic there to handle those UDT functions on-the-fly since it's a "core" feature which is independent of any specific UDT).